src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr, int ParamMaxLength)
+{
+    FILE *fPtr;
+    char buffer[ParamMaxLength + 64]; //ParamMaxLength is the size of the value, we add 64 to make sure we have enough room to store the rest of the line.
```

Comment:
I must not understand C at all, doesn't the variable get destroyed automatically when the function ends?

"buffer" is supposed to hold the contents of the entire line. ParamMaxLength is specifying the max legth of the value. So if the line is:

device = carrot

They only need to specify the max length of "carrot", but our buffer needs to hold "device = carrot"